### PR TITLE
Convert place positions to numbers

### PR DESCRIPTION
**Pull request by rock3r**
## Problem

Syx is sending us positions for places as strings instead of numbers. We need to convert them to be able to use them in the app.

## Solution

Add a conversion step as a temporary workaround until Syx fixes things on their side.

### Test(s) added

No

### Paired with 

Nobody

### CODE REVIEW

File: functions/src/events-view/mapping-functions.ts [hunk 1]
```diff
@@ -107,3 +108,13 @@ const typeFrom = (talkType: Optional<string>, track: Track | null) => {
 
     return type
 }
+
+export const convertPlaceDataToPlaceDataWithNumericPosition =
+    (places: WithId<PlaceData>[]): WithId<PlaceDataWithNumericPosition>[] => {
+        // TS is really unhappy if I inline this function in the map()
+        const convertToNumericPosition = (place: WithId<PlaceData>): WithId<PlaceDataWithNumericPosition> => {
+            return { ...place, position: Number(place.position) }
```
Comment (fourlastor) [inline functions/src/events-view/mapping-functions.ts]: I think this should be `Number.parseInt(place.position)` given it's an integer. Also I think you forgot `new` 👅

File: functions/src/firestore/data.ts [hunk 2]
```diff
@@ -50,6 +50,13 @@ export interface DayData {
 export interface PlaceData {
     readonly floor: string
     readonly name: string
+    readonly position: string
+}
+
+export interface PlaceDataWithNumericPosition {
```
Comment (fourlastor) [inline functions/src/firestore/data.ts]: I'd move this to `mapping-functions` as it's a trasformation and not something that comes from the DB

In that case we could rename it with `PlaceWithIntPosition`.

File: functions/src/events-view/mapping-functions.ts (hunk 1 above)
Comment (rock3r) [reply functions/src/events-view/mapping-functions.ts]: `Number()` is apparently [a function](https://www.w3schools.com/jsref/jsref_number.asp) but fair point on using `parseInt()`

File: functions/src/firestore/data.ts (hunk 2 above)
Comment (rock3r) [reply functions/src/firestore/data.ts]: 2bacbf1d55ec78da8c0888bec13780592ff84823

File: functions/src/events-view/mapping-functions.ts (hunk 1 above)
Comment (rock3r) [reply functions/src/events-view/mapping-functions.ts]: eea743276c6cea65f4669a0bc98ac9acd87bccc6

**Pull request merged**